crates/nargo/src/cli/mod.rs

```diff
@@ -76,12 +78,18 @@ pub fn start_cli() {
             App::new("verify")
                 .about("Given a proof and a program, verify whether the proof is valid")
                 .arg(Arg::with_name("proof").help("The proof to verify").required(true))
+                .arg(Arg::with_name("circuit_name").help(
+                    "The name of the circuit build files (ACIR, proving and verification keys)",
+                ).required(true))
                 .arg(allow_warnings.clone()),
         )
         .subcommand(
             App::new("prove")
                 .about("Create proof for this program")
                 .arg(Arg::with_name("proof_name").help("The name of the proof"))
+                .arg(Arg::with_name("circuit_name").help(
+                    "The name of the circuit build files (ACIR, proving and verification keys)",
+                ).required(true))
```

Comment:
What did you run to get this error? I would get this when not providing the `circuit_name`
```
error: The following required arguments were not provided:
    <circuit_name>

USAGE:
    nargo verify [FLAGS] <proof> <circuit_name>

For more information try --help
```
I ask mainly out of curiosity because either way I am going to change the parameter to be optional and to have a fallback

Reply:
I just ran `nargo prove` but I'd also want to be able to run `nargo prove p` which shouldn't be allowed currently afaik.

Reply:
I changed `nargo prove` back to a strategy that has the optional fallback. If a `circuit_name` is not provided we preprocess the proving and verification keys.

This is actually much better for the general `prove_and_verify` workflow executed by the noir integration tests as well. During the CI run we now avoid writing and reading from file constantly for multiple back to back tests. The old workflow requiring `nargo compile` first caused CI slow-down that I was looking at how to reduce, but now we are at the same or better speeds. The test `noir_integration` completed on my local machine in 330s, while `noir_integration` on Noir master completed in 391s. 

I have yet to update `nargo verify`, as I was considering the different options but will push something in a moment. I think for this I will also just re-run preprocess if the circuit artifacts have not been provided. Regenerating the keys on a nargo verify is technically what was happening before so this isn't going backwards. It will just be that if someone wants to get the performance benefits from not re-generating proving and verification keys, they will have to run `nargo compile` and pass a `circuit_name` to `nargo prove` and `nargo verify`.

Reply:
Awesome, thanks.